### PR TITLE
CLOUDP-72563: Add e2e tests maintenance windows

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-72563

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Comment by andreaangiolillo** [comment]
waiting for https://github.com/mongodb/mongocli/pull/430 and https://github.com/mongodb/mongocli/pull/432

File: internal/cli/cloudmanager/cloud_manager.go [hunk 1]
```diff
@@ -45,18 +46,21 @@ func Builder() *cobra.Command {
...
+		servers.Builder(),
+		automation.Builder(),
+		security.Builder(),
+		dbusers.Builder(),
+		events.Builder(),
+		processes.Builder(),
+		metrics.Builder(),
+		logs.Builder(),
+		agents.Builder(),
+		maintenance.Builder(),
```
Comment (andreaangiolillo) [inline internal/cli/cloudmanager/cloud_manager.go:62]: the importante of testing - Vol 2 😬

File: e2e/atlas/maintenance_test.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	cliPath, err := e2e.Bin()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	projectName := fmt.Sprintf("e2e-integration-proj-%v", n)
```
Comment (gssbzn) [inline e2e/atlas/maintenance_test.go:41]: copy/paste
```suggestion
	projectName := fmt.Sprintf("e2e-maintenance-proj-%v", n)
```

File: e2e/atlas/maintenance_test.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+			"--projectId",
+			projectID)
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
+
+		a := assert.New(t)
+		a.NoError(err, string(resp))
+
+		expected := "Maintenance window updated.\n"
+		a.Equal(expected, string(resp))
```
Comment (gssbzn) [inline e2e/atlas/maintenance_test.go:71]: if there's an error go will still run the nect block so we should do something like
```suggestion
		if a.NoError(err, string(resp)) {
				expected := "Maintenance window updated.\n"
				a.Equal(expected, string(resp))
		}
```

File: e2e/cloud_manager/maintenance_test.go [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	cliPath, err := e2e.Bin()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	projectName := fmt.Sprintf("e2e-integration-proj-%v", n)
```
Comment (gssbzn) [inline e2e/cloud_manager/maintenance_test.go:42]: copy/paste

File: e2e/cloud_manager/maintenance_test.go [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+			t.Errorf("error deleting project: %v", e)
+		}
+	}()
+
+	startDate := time.Now().Format(time.RFC3339)
+	endDate := time.Now().AddDate(0, 0, 1).Format(time.RFC3339)
+	var maintenanceWindowID string
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
```
Comment (gssbzn) [inline e2e/cloud_manager/maintenance_test.go:59]: [q] what error is this checking?

**Review by gssbzn — CHANGES_REQUESTED** [review]
there's a couple of small things here mostly look like copy/paste but in general this looks great, it's always great to add these

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**